ops.py

```diff
@@ -16,7 +16,7 @@ def __init__(self, *args, tensor_type, tensor_shape, patches=[], **kwargs):
         self.tensor_shape = tensor_shape
         self.patches = patches
 
-    def __new__(cls, *args, tensor_type, tensor_shape, patches=[], **kwargs):
+    def __new__(cls, *args, tensor_type, tensor_shape, tensor_data=None, patches=[], **kwargs):

```

Comment:
This is probably left over from testing

Reply:
you're correct. when i didn't understand why the conv stuff was using the tensor data, i tried storing it separately.

i will fix this stuff and clean it up. i was just holding off to wait and see if you were interested in the approach in general/if this had a chance of being merged before i put more time into it. i'll probably get a chance to work on it tomorrow.

you probably already know this, but you can edit the branch a pull targets before merging it. i've found it useful to basically take ownership of a pull that way: create a branch, merge the person's pull into it and then i can make my own changes (and possibly own pull request) before i merge it into the main branch and they'll still get appropriate credit. it may make sense to do that after i've completed most of my changes. since you seem to have a better grasp of how to detect the model types/adjust the quant heuristics, it might be easier for you to just do yourself rather than directing me.

just a possible approach, definitely don't have to do it that way or anything.

Reply:
>i will fix this stuff and clean it up. i was just holding off to wait and see if you were interested in the approach in general/if this had a chance of being merged before i put more time into it. i'll probably get a chance to work on it tomorrow.

Oh yeah, I definitely think this is worth merging now, and in hindsight I do agree that your solution of using the metadata in the gguf file is the less hacky approach. If we ever find a cleaner way due to some changes upstream we can still support loading both as it's not a massive overhead or anything.

You did a great job, the 1GB VRAM crowd will love this lol.

![image](https://github.com/user-attachments/assets/068813de-175c-4ada-a6ea-bc13952109bb)

>you probably already know this, but you can edit the branch a pull targets before merging it.

Yeah just figured it'd be easier if you made all the changes you wanted/had planned first but I'm open to doing either. You can leave the model logic out and ping me when you're done if you'd prefer that.

Reply:
> Yeah just figured it'd be easier if you made all the changes you wanted/had planned first

yes, absolutely, that's what i meant. not right now, but possibly after i've completed my changes and cleaned stuff up.

> You did a great job, the 1GB VRAM crowd will love this lol.

thanks!

Reply:
Sounds good, no rush, we may also want to wait for upstream llama.cpp for that FP32 bug mentioned below if we don't want to work around it.

Once it's all done I'll pop some example models like SDXLv0.9 and SD1.2 on HF to show the general size reduction while encouraging people to make their own quants with actually useful models.